plugins/modules/cloud/xenserver/xenserver_guest.py

```diff
@@ -104,56 +103,114 @@
   hardware:
     description:
     - Manage VM's hardware parameters. VM needs to be shut down to reconfigure these parameters.
-    - 'Valid parameters are:'
-    - ' - C(num_cpus) (integer): Number of CPUs.'
-    - ' - C(num_cpu_cores_per_socket) (integer): Number of Cores Per Socket. C(num_cpus) has to be a multiple of C(num_cpu_cores_per_socket).'
-    - ' - C(memory_mb) (integer): Amount of memory in MB.'
     type: dict
+    suboptions:
+      num_cpus:
+        description: Number of CPUs.
```

Comment:
This is just a suggestion, not a request. In my code I use this style:

```
suboptions:
  num_cpus:
    description:
    - Number of CPUs.
```

Same for other one line descriptions.